com.unity.render-pipelines.high-definition-config/Runtime/ShaderConfig.cs

```diff
@@ -51,32 +78,55 @@ public enum ShaderOptions
         // Edit->Render Pipeline->Generate Shader Includes
         // Probe Volumes feature must also be enabled inside of your HDRenderPipelineAsset.
         // Also uncomment in the HDRP package all ".../Experimental/Probe Volume" menu
+
+        /// <summary>Probe volume evaluation mode.</summary>
         ProbeVolumesEvaluationMode = ProbeVolumesEvaluationModes.Disabled,
+        /// <summary>Probe volume supports additive blending.</summary>
         ProbeVolumesAdditiveBlending = 1,
+        /// <summary>Probe volume filtering mode.</summary>
         ProbeVolumesBilateralFilteringMode = ProbeVolumesBilateralFilteringModes.Validity,
+        /// <summary>Probe volume encoding.</summary>
         ProbeVolumesEncodingMode = ProbeVolumesEncodingModes.SphericalHarmonicsL1,
 
+        /// <summary>Support for area lights.</summary>
         AreaLights = 1,
 
+        /// <summary>Support for barn doors.</summary>
         BarnDoor = 0
     };
 
     // Note: #define can't be use in include file in C# so we chose this way to configure both C# and hlsl
     // Changing a value in this enum Config here require to regenerate the hlsl include and recompile C# and shaders
+    /// <summary>
+    /// Project wide shader configuration options.
+    /// This class reflects the ShaderOptions enum. It's meant to be used in C# code to check current configuration.
+    /// </summary>
     public class ShaderConfig
     {
-        public const int k_XRMaxViewsForCBuffer = 2;         // REALLY IMPORTANT! This needs to be the maximum possible XrMaxViews for any supported platform!
-                                                             // this needs to be constant and not vary like XrMaxViews does as it is used to generate the cbuffer declarations
+        // REALLY IMPORTANT! This needs to be the maximum possible XrMaxViews for any supported platform!
+        // this needs to be constant and not vary like XrMaxViews does as it is used to generate the cbuffer declarations
+        /// <summary>Maximum number of XR views for constant buffer allocation.</summary>
+        public const int k_XRMaxViewsForCBuffer = 2;
 
+        /// <summary>Use camera relative rendering to enhance precision.</summary>
         public static int s_CameraRelativeRendering = (int)ShaderOptions.CameraRelativeRendering;
+        /// <summary>Use pre-exposition to enhance color precision.</summary>
         public static int s_PreExposition = (int)ShaderOptions.PreExposition;
+        /// <summary>Maximum number of views for XR.</summary>
         public static int s_XrMaxViews = (int)ShaderOptions.XrMaxViews;
+        /// <summary>Precomputes atmospheric attenuation for the directional light on the CPU, which makes it independent from the fragment's position, which is faster but wrong.</summary>
```

Comment:
Indicates whether to precompute atmosphere attenuation for the directional light on the CPU.